src/Expression.php

```diff
@@ -522,13 +522,9 @@ public function execute(object $connection = null)
                             ->addMoreInfo('type', gettype($val));
                     }
 
-                    // workaround to support LOB data type 2/2, see https://github.com/doctrine/dbal/pull/2434
+                    // workaround to support LOB data type 2/3, see https://github.com/doctrine/dbal/pull/2434
                     $statement->boundValues[$key] = $val;
-                    if ($type === \PDO::PARAM_STR) {
-                        $bind = $statement->bindParam($key, $statement->boundValues[$key], $type, strlen((string) $val));
-                    } else {
-                        $bind = $statement->bindParam($key, $statement->boundValues[$key], $type);
-                    }
+                    $bind = $statement->bindParam($key, $statement->boundValues[$key], $type);
```

Comment:
## fix bindParam in Expression (at least for for MSSQL) ##
see https://social.msdn.microsoft.com/Forums/biztalk/en-US/c7431849-8594-4da2-bef6-8702b72391cc/php-pdo-bindparam-optional-length-parameter-problem#answers

lenght should be set only if param is read back (in-out), not our case